Source/Immutable/Private/Immutable/ImmutablePassport.cpp

```diff
@@ -568,48 +545,42 @@ void UImmutablePassport::OnConnectPKCEResponse(FImtblJSResponse Response) {
 }
 #endif
 
-void UImmutablePassport::OnGetIdTokenResponse(FImtblJSResponse Response) {
-  if (auto ResponseDelegate = GetResponseDelegate(Response)) {
-    // Extract the credentials
-    auto Credentials =
-        JsonObjectToUStruct<FImmutablePassportTokenData>(Response.JsonObject);
+void UImmutablePassport::OnGetIdTokenResponse(FImtblJSResponse Response)
+{
+  if (auto ResponseDelegate = GetResponseDelegate(Response))
+  {
+    if (!Response.success || !Response.JsonObject->HasField(TEXT("result")))
+    {
+      IMTBL_LOG("No stored Id token found.");
 
-    if (!Response.success || !Credentials.IsSet() ||
-        !Credentials->idToken.Len()) {
-      IMTBL_LOG("No stored credentials found.");
-      FString Msg;
-      Response.Error.IsSet()
-          ? Msg = Response.Error->ToString()
-          : Msg = Response.JsonObject->GetStringField(TEXT("error"));
-      ResponseDelegate->ExecuteIfBound(
-          FImmutablePassportResult{false, Msg, Response});
-    } else {
-      IMTBL_LOG("Stored ID token found.");
-      ResponseDelegate->ExecuteIfBound(
-          FImmutablePassportResult{true, Credentials->idToken});
+      const FString Msg = Response.JsonObject->HasField(TEXT("error")) ? Response.JsonObject->GetStringField(TEXT("error")) : "Failed to retrieve Id Token.";
+
+      ResponseDelegate->ExecuteIfBound(FImmutablePassportResult{false, Msg, Response});
+    }
+    else
+    {
+      IMTBL_LOG("Stored Id token found.");
+      ResponseDelegate->ExecuteIfBound(FImmutablePassportResult{true, Response.JsonObject->GetStringField(TEXT("result"))});
     }
   }
 }
 
-void UImmutablePassport::OnGetAccessTokenResponse(FImtblJSResponse Response) {
-  if (auto ResponseDelegate = GetResponseDelegate(Response)) {
-    // Extract the credentials
-    auto Credentials =
-        JsonObjectToUStruct<FImmutablePassportTokenData>(Response.JsonObject);
+void UImmutablePassport::OnGetAccessTokenResponse(FImtblJSResponse Response)
+{
+  if (auto ResponseDelegate = GetResponseDelegate(Response))
+  {
+    if (!Response.success || !Response.JsonObject->HasField("result"))
```

Comment:
I think we should also check that the `result` is not empty.
Same for OnGetIdTokenResponse and HasStoredCredentials

Reply:
Fixed by using TryGet... methods